src/Rules/ModelProperties/ModelPropertiesRuleHelper.php

```diff
@@ -0,0 +1,215 @@
+
+class ModelPropertiesRuleHelper
+{
+    /** @var ModelPropertyExtension */
+    private $modelPropertyExtension;
+
+    /** @var AnnotationsPropertiesClassReflectionExtension */
+    private $annotationExtension;
+
+    /** @var ModelAccessorExtension */
+    private $modelAccessorExtension;
+
+    public function __construct(ModelPropertyExtension $modelPropertyExtension, ModelAccessorExtension $modelAccessorExtension, AnnotationsPropertiesClassReflectionExtension $annotationExtension)
+    {
+        $this->modelPropertyExtension = $modelPropertyExtension;
+        $this->modelAccessorExtension = $modelAccessorExtension;
+        $this->annotationExtension = $annotationExtension;
+    }
+
+    /**
+     * @param MethodReflection $methodReflection
+     * @param Scope            $scope
+     * @param Node\Arg[]       $args
+     * @param ClassReflection|null  $modelReflection
+     *
+     * @return string[]
+     */
+    public function check(MethodReflection $methodReflection, Scope $scope, array $args, ?ClassReflection $modelReflection = null): array
+    {
+        $modelPropertyParameter = $this->hasModelPropertyParameter($methodReflection, $scope, $args, $modelReflection);
+
+        if (count($modelPropertyParameter) !== 2) {
+            return [];
+        }
+
+        /** @var int $parameterIndex */
+        /** @var ObjectType $modelType */
+        [$parameterIndex, $modelType] = $modelPropertyParameter;
+
+        $modelReflection = $modelType->getClassReflection();
+
+        if ($modelReflection === null) {
+            return [];
+        }
+
+        if ($modelReflection->isAbstract()) {
+            return [];
+        }
+
+        if ($modelReflection->getName() === Model::class || ! $modelReflection->isSubclassOf(Model::class)) {
+            return [];
+        }
+
+        if (! array_key_exists($parameterIndex, $args)) {
+            return [];
+        }
+
+        $argValue = $args[$parameterIndex]->value;
+
+        if (! $argValue instanceof Node\Expr) {
+            return [];
+        }
+
+        $argType = $scope->getType($argValue);
+
+        if ($argType instanceof ConstantArrayType) {
+            $errors = [];
+
+            $keyType = TypeUtils::generalizeType($argType->getKeyType());
+
+            if ($keyType instanceof IntegerType) {
+                $valueTypes = $argType->getValuesArray()->getValueTypes();
+            } elseif ($keyType instanceof StringType) {
+                $valueTypes = $argType->getKeysArray()->getValueTypes();
+            } else {
+                $valueTypes = [];
+            }
+
+            foreach ($valueTypes as $valueType) {
+                // It could be something like `DB::raw`
+                // We only want to analyze strings
+                if (! $valueType instanceof ConstantStringType) {
+                    continue;
+                }
+
+                // TODO: maybe check table names and columns here. And for JSON access maybe just the column name
```

Comment:
I think a simple:
```php
$columnName = Str::after($valueType->getValue(), '.');
```

would be nice here (and below). So that we can also check the column name if it contains a table prefix.

Reply:
I don't understand. We are already checking in the next line if the value contains a dot. There is no point of getting the substring after the dat and checking again.

Reply:
Ah, I misunderstood I guess. You are suggesting a fix to this TODO. 

Yeah, we can do that. But then we need to be able to check the column in the given table, not the model we know. So it has to ask to `MigrationHelper` if that table exists and if that column is in that table.

Reply:
Ah for a second I thought ignoring the table would be enough here. But that's not the case when tables have been joined together.